### PR TITLE
Add "glimmer-component" blueprint to the project

### PR DESCRIPTION
**Pull request by Turbo87**
Enables us to use "ember generate glimmer-component foo-bar" in projects that depend on "glimmer-application-pipeline"

/cc @tomdale

### CODE REVIEW

File: blueprints/glimmer-component/index.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ module.exports = {
   },
 
   locals(options) {
-    let className = stringUtils.classify(options.entity.name);
+    let className = `${stringUtils.classify(options.entity.name)}Component`;
```
Comment (locks) [inline blueprints/glimmer-component/index.js]: not sure I'm a fan of this, as it brings to mind imagines of ye olde global resolver :P

**Comment by dgeb** [comment]
Thanks for getting this started. I just talked with @rwjblue and @tomdale - could you please make the following changes: 

* Change the blueprint from `glimmer-component` to just plain `component`
* Leave the name of the class out of the module (i.e. `export default class extends Component {`)

**Comment by Turbo87** [comment]
> Leave the name of the class out of the module (i.e. export default class extends Component {)

yes, no problem

> Change the blueprint from glimmer-component to just plain component

this is a problem. we are currently hardcoding the `ember-cli-legacy-blueprints` addon into every project and I would have to override the `component` blueprint coming from that addon. That in itself is not a problem, but the addon also provides a `component-test` blueprint which is automatically used when the `component` blueprint is used. I could override that blueprint too with an empty one, but it would still display `installing component-test` on the command line.

I've updated the PR with that approach, but I'm not 100% happy with it :-/

**Comment by dgeb** [comment]
Thanks @Turbo87. 

@tomdale raised the point that naming the class will improve debugging, so maybe removing wasn't what we wanted after all. If we are keeping the name, we will need to decide whether it keeps a `Component` suffix (as was done originally) or leaves it off (which seems cleaner to me).

@rwjblue - could you weight in on @Turbo87's concerns re: `ember-cli-legacy-blueprints`? Is there a path to not including those at all in glimmer apps, esp. considering that we're using ember-cli canary?

**Comment by rwjblue** [comment]
Well, clearly we want tests to "be a thing" in the long run so it isn't super bad that we have a blueprint for them here. Also, ember-cli should start down a path to allow us to remove the legacy blueprints package. I'll think about this, and try to circle around with Tobias this morning. 

RE: debugging and naming the class, I am not convinced. All of the normal ember-cli blueprints follow the pattern I described (`export default class {}` without giving that thing a name), so clearly it isn't super bad. Please keep in mind that one of the points of MU was that things were "portable", and embedding the name inside the file contents breaks that...

**Comment by Turbo87** [comment]
> a path to allow us to remove the legacy blueprints package

I was thinking maybe a flag in `package.json` that allows addons to disable the legacy blueprints addon for the host app. That flag could then be set by `ember-source` and this project right here.

> RE: debugging and naming the class

we could in theory also add a Babel plugin or something like that which adds the name automatically if the class is anonymous 😉 

the difference to the previous situation is that now we use ES6 classes, so the naming would probably be a bit more useful for debugging

**Comment by dgeb** [comment]
> I was thinking maybe a flag in package.json that allows addons to disable the legacy blueprints addon for the host app.

Something like that seems good to me. Let me know what you guys decide.

> we could in theory also add a Babel plugin or something like that which adds the name automatically if the class is anonymous

Yeah, that's clever 👍  How about if we keep the class name (without the `Component` suffix) so we get the advantages for debugging ES classes, and then we remove the class name once we can auto-generate it. @rwjblue - does that work for you?

**Comment by dgeb** [comment]
@Turbo87 @rwjblue have you had a chance to strategize on how to disable legacy blueprints?

**Comment by Turbo87** [comment]
not yet, unfortunately

**Comment by Turbo87** [comment]
I just tried to spike this out, but it seems a bit more complicated than I hoped since the legacy blueprint addon filtering needs to happen during addon discovery, at which point the addons are not available yet... sort of a chicken and egg situation

**Comment by dgeb** [comment]
Superceded by #30

**Pull request closed without merging**